tf2/include/tf2/exceptions.h

```diff
@@ -39,15 +39,18 @@
 
 namespace tf2{
 
+#ifndef TF2ERROR_ENUM
+#define TF2ERROR_ENUM
 enum class TF2Error : std::uint8_t {
-  NO_ERROR = 0,
+  NO_ERRORS = 0,
```

Comment:
Not sure if this is a requirement but from my understanding, this should match the enums defined in the corresponding TF2Error ROS message https://github.com/ros2/geometry2/blob/fe1858c8155d94d52f3591529379d29192754cb0/tf2_msgs/msg/TF2Error.msg#L1.

For future references: This seems related to existing issues about windows defines: https://github.com/ros/geometry2/issues/172 and https://github.com/ros2/rosidl/issues/118.

Reply:
Thanks!  I was blissfully unaware of that.  Have pushed a fix to match the message and the enum.

Reply:
thanks! I'll leave it up to @tfoote to review the rest of this PR and comment on the best way to move forward regarding the enum names as he is the one that has the most context.